### PR TITLE
git-lfs: add v3.4.1, v3.5.1

### PR DESCRIPTION
**Pull request by wdconinc**
This PR adds the versions 3.4.1 and 3.5.1 of git-lfs, and their dependencies on the respective versions of Go as called out in the changelog (git-lfs supports the latest version of Go, per https://github.com/git-lfs/git-lfs/blob/a3829f486597702297737bb6677534cb5cd0bbc6/CONTRIBUTING.md#prerequisites, and older versions are dropped from tests in CI, e.g. https://github.com/git-lfs/git-lfs/pull/5487).

Test build:
```
[+] /opt/software/linux-ubuntu24.04-skylake/gcc-13.2.0/git-lfs-3.5.1-jxoaatqjalwkacomfys7tg5yblbr2leo
```

### CODE REVIEW

**Review by alecbcs — APPROVED** [review]
Verified Checksums. Looks good to me. Waiting on final approval from @sethrj.

**Review by sethrj — APPROVED** [review]

**Pull request merged**
